examples/migration/campaign_management/create_complete_campaign_both_apis_phase_1.rb

```diff
@@ -0,0 +1,280 @@
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This code example is the second in a series of code examples that shows how to create
+# a Search ad_group using the AdWords API, and then migrate it to the Google Ads API one
+# functionality at a time. See other examples in this directory for code examples in various
+# stages of migration.
+#
+# In this code example, the functionality to create ad_group budget has been migrated to
+# the Google Ads API. The rest of the functionality - creating a Search ad_group, ad groups,
+# keywords and expanded text ads are done using the AdWords API.
+
+
+require 'optparse'
+require 'google/ads/google_ads'
+require 'adwords_api'
+require 'date'
+
+def add_complete_campaign(customer_id)
+  # AdwordsApi::Api will read a config file from ENV['HOME']/adwords_api.yml
+  # when called without parameters.
+  adwords = AdwordsApi::Api.new
+
+  # To enable logging of SOAP requests, set the log_level value to 'DEBUG' in
+  # the configuration file or provide your own logger:
+  # adwords.logger = Logger.new('adwords_xml.log')
+
+  # GoogleAdsClient will read a config file from
+  # ENV['HOME']/google_ads_config.rb when called without parameters
+  # Please note: this client uses the new Google Ads API, and we share both
+  # Ads and AdWwords clients in process
+  client = Google::Ads::GoogleAds::GoogleAdsClient.new
+
+  budget_id = create_budget(client, customer_id)
+  campaign_id = create_campaign(adwords, budget_id)
+  ad_group_id = create_ad_group(adwords, campaign_id)
+
+  create_text_ads(adwords, ad_group_id)
+  create_keywords(adwords, ad_group_id, KEYWORDS_TO_ADD)
+end
+
+def create_budget(client, customer_id)
+  # Create a budget, which can be shared by multiple campaigns.
+  budget = client.resource(:CampaignBudget)
+  budget.name = client.wrapper.string(
+    "Interplanetary cruise budget ##{(Time.new.to_f * 1000).to_i}",
+  )
+  budget.amount_micros = client.wrapper.int64(
+    50_000_000,
+  )
+
+  budget.delivery_method = :STANDARD
+  operation = client.operation(:CampaignBudget)
+  operation["create"] = budget
+
+  campaign_budget_srv = client.service(:CampaignBudget)
+  response = campaign_budget_srv.mutate_campaign_budgets(customer_id, [operation])
+  budget_resource_name = response.results.first.resource_name
+
+  puts("Created campaign budget with resource name #{budget_resource_name}")
+
+  # IDs are given by the last slash in the resource name, so this converts
```

Comment:
"IDs are given by the last slash in the resource name"

You probably copied this comment from the golden but it's not entirely accurate. Some resource names don't have anything in this 4th position, and some have multiple IDs or other unique identifiers concatenated together. The logic to extract a given ID is going to vary depending on what type of entity it is and what ID you want.